### ISSUE TITLE
Enqueue CSS and JS on Non-Content Sections

### ISSUE THREAD

**Issue by jmbarne3** [UCF/UCF-Section-Plugin#29]
When a section is put into a customizer setting, widget content or ACF field (anything other than the `post_content`), the CSS and JS is not currently being enqueued. This is because the function that looks for sections in order to enqueue the files can't be aware of everywhere it needs to look for sections. We should add a filter or action that can be used on themes to add section, and its corresponding css and js to the global fields for enqueuing.

**Comment by jmbarne3** [comment]
Idea:

Create a `get_post_sections` filter that passes in the sections already found by the default logic. Within this plugin, we refactor the code a little bit so that there's a single function that can be called that accepts the content as a string and returns whatever sections it finds. The resulting function within the _themes_ would then look something like...

```
function theme_pre_get_post_sections( $sections ) {
   $content = function_that_gets_the_content_we_need_to_process();
   $additional_sections = UCF_Section_Common::find_post_sections( $content ); // or whatever we call this function...
   return array_merge( $sections, $additional_sections );
}

add_filter( 'get_post_sections', 'theme_pre_get_post_sections', 10, 1 );
```

The default logic will then take care of adding whatever CSS and JS need to be enqueued.

**Comment by jmbarne3** [comment]
Solved in #30
